### ISSUE TITLE
Support routing/v1alpha2 TCP routes

### ISSUE THREAD

**Issue by ijsnellf** [istio/istio#2468]
Support TCP routing via the v1alpha2 routing specification.

**Comment by ijsnellf** [comment]
Contributes to https://github.com/istio/istio/issues/1925

**Comment by mydoal** [comment]
Are the TCP Routing rules implemented via the VirtualService already in prelim 0.8?

**Comment by louiscryan** [comment]
@rshriram this would be launch blocking for 1.0 if not already resolved.

**Comment by jasmine-jaksic** [comment]
@ijsnellf Are you working on this? If so, can you provide an update (and ETA)?

**Comment by rshriram** [comment]
A big blocker for this is supporting destination/source ip/cidr matching in Envoy. Cc @PiotrSikora
We have some basic stuff out for a PR though

**Comment by jasmine-jaksic** [comment]
Update: ETA for fix end of week

**Comment by jasmine-jaksic** [comment]
@PiotrSikora @rshriram Any progress on this?

**Comment by PiotrSikora** [comment]
@jasmine-jaksic yeah, but not done yet... I hope to have the Envoy PR out later tonight, though.

**Comment by hklai** [comment]
@PiotrSikora is expecting to merge the fix by tomorrow.

**Comment by ijsnellf** [comment]
Major pieces remaining:

- [ ] Better wildcard handling. Virtual services and gateway servers with wildcard hosts should be matched from most specific to least specific. (So, if we match both `*.uk.com` and `*`, we choose `*.uk.com` because it is more specific.)
- [ ] Arbitrary server names in the TLS match. Currently, the server name must have an associated service in the registry.

**Comment by PiotrSikora** [comment]
> Better wildcard handling. Virtual services and gateway servers with wildcard hosts should be matched from most specific to least specific. (So, if we match both *.uk.com and *, we choose *.uk.com because it is more specific.)

That should be already happening if you use filter chain matching.

**Comment by rshriram** [comment]
>Arbitrary server names in the TLS match. Currently, the server name must have an associated service in the registry.
 
??? I don’t remember such a restriction.

**Comment by louiscryan** [comment]
@ijsnellf this bug is becoming a bit of an umbrella. Can you create more focused bugs for the remaining issues and close this one. Please attach to the 1.1 milestone for triage.

**Comment by zx3821ren** [comment]
I find sourceSubnet in L4MatchAttributes removed in Istio 1.0 doc. 
Will the sourceSubnet be support or complete L4 connection matching support in 1.1? And When 1.1 released? We need to do ip white listing with istio for tcp route.

**Comment by zx3821ren** [comment]
@rshriram  Any Update for the issue?
